cli/js/compiler.ts

```diff
@@ -71,9 +74,29 @@ interface CompileResult {
   diagnostics?: Diagnostic;
 }
 
+// this is used to generate the foundational AST for all other compilations, so
+// it can be cached as part of the
+let oldProgram: ts.Program;
+
+((): void => {
+  const ops = core.ops();
+  for (const [name, opId] of Object.entries(ops)) {
+    const opName = `OP_${name.toUpperCase()}` as keyof typeof dispatch;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (dispatch as any)[opName] = opId;
+  }
+  const host = new Host({ writeFile(): void {} });
+  const options = host.getCompilationSettings();
+  oldProgram = ts.createProgram({
+    rootNames: [`${ASSETS}/example.ts`],
```

Comment:
Is this hardcoded `example.ts` okay?

Reply:
We need something to exist as a root name... an empty value does not work.  Totally open to something else.

Reply:
Oh so this is just a placeholder? It's only job is to load type definitions of Deno and all other assets? If so, a comment will suffice